core-utilities/src/main/java/edu/illinois/cs/cogcomp/annotation/BasicAnnotatorService.java

```diff
@@ -371,7 +371,12 @@ public TextAnnotation addViewsAndCache(TextAnnotation ta, Set<String> viewsToAnn
                 ta = annotationCache.getTextAnnotation(ta);
 
         for (String viewName : viewsToAnnotate) {
-            isUpdated = addView(ta, viewName) || isUpdated;
+            try {
+                isUpdated = addView(ta, viewName) || isUpdated;
+            } catch (AnnotatorException e) {
+                // the exception is handled here, because one single view failure should not resutl in loss of all the annotations
```

Comment:
Personally, I do not like this being the default behavior for handling errors. We should defined a runtime parameter for the user to explicitly choose this behavior. Let me know what you think?

Aside: This will also affect downstream MapDB caching if the TA is cached using hash of text, corpus Id, or incorrect set of views.